### ISSUE TITLE
Navbar in "mobile mode" doesn't collapse when option clicked

### ISSUE THREAD

**Issue by Scalpel78** [bootstrap-vue/bootstrap-vue#712]
Hi, 
I'm just getting started with bootstrap-vue, so this might be something I'm doing wrong.

I'm using the code from https://bootstrap-vue.js.org/docs/components/navbar on my page. When I shrink the horizontal size if the page, so that it is displayed as on a mobile, click the hamburger menu, and then click any option I expected the navbar to toggle away. For the user it seems like nothing happens when clicking the menu item.

I went to getbootstrap.com to see how it behaved there, and it has the same behavior, so I'm guessing that Bootstrap and Bootstrap-Vue handles this in the same way.

How can I tell the navbar to de-toggle when the user clicks an item when in "mobile size"?

**Comment by tmorehouse** [comment]
Hmmm that maybe in an issue with how they implement the collapse sub-component.

We might be able to get around this by adding a click handler on the collapse portion of the nav, but we would need to know which state the toggle is in.

**Comment by tmorehouse** [comment]
I think I have determined a method for handling this, by detecting the computed display mode of hte collapse when in `is-navbar` mode, and adding a `click` handler to the wrapper component of the collapse.

**Comment by tmorehouse** [comment]
I have another PR that tweaks the collapse component.  Once it is merged I will work on this issue and hopefully have it available for release v0.19.0

**Comment by Scalpel78** [comment]
Super, looking forward to testing that version.

**Comment by tmorehouse** [comment]
PR #714 should close the collapsable navbar when clicked.

While this deviates from the current bootstrap V4.alpha.6 behaviour, it does provide a better end user experience.

**Comment by petr-buchin** [comment]
@tmorehouse I see the typo in this merge request:
https://github.com/bootstrap-vue/bootstrap-vue/pull/714/files#diff-e4db3a7cc0eb7e2a471aed3656287b6eR148

Method in line 148 has name `handleReszie()` while it is referenced in lines 166 and 174 as `this.handleResize`. I think this should be fixed.

**Comment by tmorehouse** [comment]
This has already been fixed in the latest master via commit 19f498a9c0528f6e66ee51573452954bfeefe5d2

**Comment by petr-buchin** [comment]
Oh, thanks :)

**Comment by tmorehouse** [comment]
This feature will be available in release v0.19.0

**Comment by petr-buchin** [comment]
@tmorehouse , I updated to the new v0.19.0 release and this fix does not seem to work :(

**Comment by tmorehouse** [comment]
Try release v0.20.0 (v0.19.0 had a few bugs in some components), and let us know

**Comment by petr-buchin** [comment]
@tmorehouse still does not work.

**Comment by tmorehouse** [comment]
OK, looking into this...

**Comment by tmorehouse** [comment]
This appears now to be working in version v0.20.2 in the official docs:

https://bootstrap-vue.js.org/docs/components/navbar

Jut make your browser window narrow in order to test it out.

**Comment by petr-buchin** [comment]
@tmorehouse after update navbar is collapsed in desktop mode, though it should be collapsed only in mobile mode.

**Comment by talalzaki** [comment]
same here

**Comment by tmorehouse** [comment]
@talalzaki which version of Bootstrap-Vue are you using?

The latest version is v1.1.0, which is based on Bootstrap V4.beta.2 CSS (the v0.xx.xx releases are now obsoleted)

**Comment by talalzaki** [comment]
i use 1.1.10 with tether 1.4.1 "bootstrap": "^4.0.0-alpha.6",

**Comment by tmorehouse** [comment]
The alpha branch has been deprecated and is no longer supported.  The last version of the Bootstrap-Vue alpha (Bootstrap-V4.alpha.6 CSS) branch is v0.23.1.

**Comment by talalzaki** [comment]
thanks its works

**Comment by saintplay** [comment]
when I have a slot, clicking on the child doesn't close the toggle

```html
<b-nav-item href="#hero" class="px-t-3">
   <span class="nav-link-text">
     start
   </span>
</b-nav-item>
```

Yes, it's absolutely necessary to have a `span`  inside the `b-nav-item`
